Sources/TuistDoc/SwiftDocServer.swift

```diff
@@ -0,0 +1,130 @@
+import Dispatch
+import Foundation
+import Signals
+import Swifter
+import TSCBasic
+import TuistSupport
+
+public protocol SwiftDocServing {
+    /// Base url for the server
+    var baseURL: String { get }
+
+    /// Serves the documentation at a given path
+    /// - Parameters:
+    ///   - path: Path to the folder containing the documentation
+    ///   - port: Port to use for hosting the website
+    func serve(path: AbsolutePath, port: UInt16) throws
+}
+
+public final class SwiftDocServer: SwiftDocServing {
+    private static let index = "index.html"
+    private static var temporaryDirectory: AbsolutePath?
+
+    /// Utility to manipulate files
+    private let fileHandling: FileHandling
+
+    /// Utility to open files
+    private let opener: Opening
+
+    /// HTTPServer from Swifter
+    private var server: HttpServer?
+
+    /// Semaphore to block the execution
+    private var semaphore: DispatchSemaphore?
+
+    public let baseURL: String = "http://localhost"
+
+    public init(fileHandling: FileHandling = FileHandler(),
+                opener: Opening = Opener())
+    {
+        self.fileHandling = fileHandling
+        self.opener = opener
+    }
+
+    public func serve(path: AbsolutePath, port: UInt16) throws {
+        SwiftDocServer.temporaryDirectory = path
+
+        func okReponse(fileAt path: AbsolutePath) throws -> HttpResponse {
+            guard let file = try? path.pathString.openForReading() else {
+                return .notFound
+            }
+            return .raw(200, "OK", [:]) { writer in
+                try? writer.write(file)
+                file.close()
+            }
+        }
+
+        func handleRequest(_ request: HttpRequest) -> HttpResponse {
+            guard let (_, value) = request.params.first else {
+                return .notFound
+            }
+
+            let filePath = path.appending(component: value)
```

Comment:
I think you can simplify the implementation significantly by using [this functionality](https://github.com/httpswift/swifter#how-to-share-files) where you can say serve all the files in this directory.

Reply:
Hey, I've tried this one but I could not make it work.

As far as I can tell, there are two reasons:
- The hrefs are generated like this: `href="http://localhost:4040/AppDelegate"` and that is a folder, not a file.
- The `shareFilesFromDirectory` function ends up downloading the files, instead of navigating.

Am I missing something?

Reply:
Oh I see. In that case, we can leave it as it is. I thought that functionality would serve our needs well.